### ISSUE TITLE
Build Numbers

### ISSUE THREAD

**Issue by PhGeek** [Block8/PHPCI#898]
Why the build numbers are based on server instead of projects?

**Comment by Adirelle** [comment]
The build number are actually build identifiers, e.g. database-generated serial numbers.

**Comment by PhGeek** [comment]
I could notice that. Actually my question is if the team intend to implement per project builds?

**Comment by Adirelle** [comment]
I don't know but I tihnk this would be a lot of changes for little added value.

**Comment by PhGeek** [comment]
Yeah it would be a lot of work but I don't agree at all. I don't see a sense in the way it works currently.
Personally it could be a reason for not choosing PHPCI.

**Comment by Adirelle** [comment]
Could you elaborate which sense build-specific numbers would make, please ?

**Comment by Libbux** [comment]
:-1:

**Comment by PhGeek** [comment]
Really? Can you understand what I'm looking for in this issue?

Numbers being incrementally related to the project. Then I'll be able to easilly determine what's going on.

Ex: What's the sense on a "Software version X.X.X build XXX" if it's not a real information? Or should I count it by hand?

PS: Sorry if it seems I'm not being polite... as I'm not a english speaker.

**Comment by zviryatko** [comment]
:+1:
Yes, @PhGeek is right. To add this feature need a few simple changes:
 1. Change build view uri in controller to something like /project/:project_id/build/:build_id (but i prefer to add machine name to project).
 2. Change primary key in database.
 3. Create migration script for backward compatibility.
 4. ...
 5. Profit!

**Comment by mikebronner** [comment]
I agree, even if you don't use major.minor.patch.build versioning, it still makes sense -- as in "This is the 5th build of project Hydrangea."

**Comment by MarkMaldaba** [comment]
It gets a bit weird when builds happen 'out of order', though, doesn't it?  For example, as may happen if builds happen in parallel (e.g. multi-threaded/distributed setups), or if manually triggered builds take priority over the automatic intermediate builds.

It also gets a bit weird when there are multiple branches of the same project, where merges go in both directions (which is common), particularly if you expect 'build order' to be the same as 'date order'.

Also, should be numbered on a 'first to start' or 'first to finish' basis?

**Comment by Adirelle** [comment]
@zviryatko: these are all but *simple* changes, and multiple field identifiers are generally awful to handle.

@MarkMaldaba: I agree with you. CI build numbers are just build identifiers in the CI system. They have no sense anyway if the project has multiple branches. What is important is the SCM tag and commit identifier.

**Comment by zviryatko** [comment]
I think for build number is good to use numbers one by one, like in jenkins. Not necessary to replace system id, but then there will be too many different id, and its bad too, like in jenkins =(

**Comment by tvbeek** [comment]
If it's only for displaying (so not for anything else like url, build path, ....) I think it can be an option that someone can implement if he/she really likes it.
If you also want to use it for other things I think it is a big risk to deliver problems now or in the future. (And remove one build already means you lost your nice line of numbers)

In short: (My opinion)
Only to display: feel free to implement
For anything else: Please don't
